File: aarthik_setu/lib/models/common/manual_itr.dart
class ManualItr{
  String firstName;
  String? middleName;
  String lastName;
  DateTime dob;
  String pan;
  String mobile;
  String email;
  String addressLine1;
  String? addressLine2;
  String landmark;
  String country;
  String pinCode;
  String state;
  String city;
  String? village;
  String? district;
  String? subDistrict;
  Map<(int,int), double> netAnnualIncome;

  ManualItr({
    required this.firstName,
    this.middleName,
    required this.lastName,
    required this.dob,
    required this.pan,
    required this.mobile,
    required this.email,
    required this.addressLine1,
    this.addressLine2,
    required this.landmark,
    required this.country,
    required this.pinCode,
    required this.state,
    required this.city,
    this.village,
    this.district,
    this.subDistrict,
    required this.netAnnualIncome,
  });

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'middleName': middleName,
      'lastName': lastName,
      'dob': dob,
      'pan': pan,
      'mobile': mobile,
      'email': email,
      'addressLine1': addressLine1,
      'addressLine2': addressLine2,
      'landmark': landmark,
      'country': country,
      'pinCode': pinCode,
      'state': state,
      'city': city,
      'village': village,
      'district': district,
      'subDistrict': subDistrict,
      'netAnnualIncome': netAnnualIncome,
    };
  }

  factory ManualItr.fromJson(Map<String, dynamic> json) {
    return ManualItr(
      firstName: json['firstName'],
      middleName: json['middleName'],
      lastName: json['lastName'],
      dob: json['dob'],
      pan: json['pan'],
      mobile: json['mobile'],
      email: json['email'],
      addressLine1: json['addressLine1'],
      addressLine2: json['addressLine2'],
      landmark: json['landmark'],
      country: json['country'],
      pinCode: json['pinCode'],
      state: json['state'],
      city: json['city'],
      village: json['village'],
      district: json['district'],
      subDistrict: json['subDistrict'],
      netAnnualIncome: json['netAnnualIncome'],
    );
  }
}

File: server/models/forms/personal_forms/itr_forms/manual_itr.go
package itr_forms

import "time"

// ManualITR represents the structure of the ITR form data
type ManualITR struct {
	FirstName      string                     `json:"firstName" binding:"required"`
	MiddleName     *string                    `json:"middleName,omitempty"`  // Optional
	LastName       string                     `json:"lastName" binding:"required"`
	DOB            time.Time                  `json:"dob" binding:"required"` // Use `time.Time` for DateTime
	PAN            string                     `json:"pan" binding:"required"`
	Mobile         string                     `json:"mobile" binding:"required"`
	Email          string                     `json:"email" binding:"required"`
	AddressLine1   string                     `json:"addressLine1" binding:"required"`
	AddressLine2   *string                    `json:"addressLine2,omitempty"` // Optional
	Landmark       string                     `json:"landmark" binding:"required"`
	Country        string                     `json:"country" binding:"required"`
	PinCode        string                     `json:"pinCode" binding:"required"`
	State          string                     `json:"state" binding:"required"`
	City           string                     `json:"city" binding:"required"`
	Village        *string                    `json:"village,omitempty"`      // Optional
	District       *string                    `json:"district,omitempty"`     // Optional
	SubDistrict    *string                    `json:"subDistrict,omitempty"`  // Optional
	NetAnnualIncome map[IncomeRange]float64    `json:"netAnnualIncome" binding:"required"`
}

// IncomeRange represents the start and end year for income records
type IncomeRange struct {
	StartYear int `json:"startYear" binding:"required"`
	EndYear   int `json:"endYear" binding:"required"`
}
